vite.config.ts

```diff
@@ -1,13 +1,27 @@
-const path = require("path");
-const { defineConfig } = require("vite");
-const commonjs = require("@rollup/plugin-commonjs");
+import { defineConfig, UserConfig } from "vite";
+import { InlineConfig } from "vitest";
+import path from "path";
+import dts from "vite-plugin-dts";
+import commonjs from "@rollup/plugin-commonjs";
 
-module.exports = defineConfig({
+interface VitestConfigExport extends UserConfig {
+  test: InlineConfig;
+}
+
+export default defineConfig({
+  test: {
+    environment: "jsdom",
+    globals: true,
+    setupFiles: ["./vitest.setup.js"],
+    deps: {
+      fallbackCJS: true,
+    },
+  },
   build: {
     minify: false,
     sourcemap: true,
     lib: {
-      entry: path.resolve(__dirname, "source/index.js"),
+      entry: path.resolve(__dirname, "source/index.ts"),
       name: "ftrack-javascript-api",
```

Comment:
the `name` here should be something like `FtrackApi`, which is the exposed global variable.

Reply:
the `name` is only used for the build output files, which are named `ftrack-javascript-api.umd.js` or `ftrack-javascript-api.es.js`. the name of the exposed variable for the umd file is `ftrack`, which is mapped under `rollupOptions.global`